src/core/raster/qgsrasterlayer.cpp

```diff
@@ -2036,6 +2036,22 @@ QString QgsRasterLayer::encodedSource( const QString &source, const QgsReadWrite
       }
     }
   }
+  else if ( providerType() == "wms" )
```

Comment:
Not your fault, but this provider-specific logic sitting inside QgsRasterLayer is horrible. Hopefully @PeterPetrik's recent work might lead eventually to a nicer solution for this...